editoast/src/views/infra/railjson.rs

```diff
@@ -239,11 +243,13 @@ mod tests {
             .uri("/infra/railjson?name=post_railjson_test")
             .set_json(&railjson)
             .to_request();
-        let response = call_service(&app, req).await;
-        assert_eq!(response.status(), StatusCode::OK);
-        let res: PostRailjsonResponse = read_body_json(response).await;
 
-        let conn = &mut db_pool.get().await.unwrap();
-        assert!(Infra::delete_static(conn, res.infra).await.unwrap());
+        let res: PostRailjsonResponse = app.fetch(req).assert_status(StatusCode::OK).json_into();
+
+        assert!(
+            Infra::delete_static(db_pool.get_ok().deref_mut(), res.infra)
+                .await
+                .unwrap()
+        );
```

Comment:
I'm not entirely sure, do we want to test for deletion OR do we NEED to delete? What happens if, for some reason, the test panics just above, does the DB stay in a clumsy situation?

Reply:
I think we're testing the delete after creation. If the test panics, we won't have any issues with in the DB since we don't commit the changes.